### ISSUE TITLE
rez tools visibility behavior not backwards compatible from 2.61.0

### ISSUE THREAD

**Issue by maxnbk** [nerdvegas/rez#1300]
between 2.60.1 and 2.61.0 , 
`src/rez/system.py` changed it's determination of "whether to pass rez-tools along to the environment or not" from being a "is it `which`-able, to, "is the currently importable rez module also a production install".

This does not have to be the case, and makes sense not to be the case. In our production environment, we have a production install of rez tools exposed on the PATH, and a separate version of rez exposed as a module (via a package), and the module is not a production install, while the one on the PATH is.

When migrating our rez-tools versions to newer versions of rez, nothing changes, but when we migrate the package-version of rez to newer versions to 2.60.0+ , the module version of rez, when performing a solve, no longer exposes the rez tools from the PATH to the downstream resolved context.

This occurred here, although subsequent edits occur to expand the logic to lib64, etc.
https://github.com/nerdvegas/rez/compare/2.60.1...2.61.0
"[-update to system.rez_bin_path that is more reliable](https://github.com/nerdvegas/rez/commit/c707e1deea8a4d6b4dfcdc6b54b50c6d1a453dfc)"
(in `./src/rez/system.py`)

**Environment**
* CentOS-7.6
* Encountered when migrating components from rez-2.52 to 2.100+
* rez-python-2.7

**To Reproduce**
1. Expose rez tools via PATH
2. rez-env to a rez-package
3. Use that rez-package's api to create a ResolvedContext and execute the shell
4. After executing the shell, inspect the PATH and find that the rez-tools are no longer available.

**Expected behavior**
I expect the rez-tools to still be available after the execution of the RC's shell.

**Actual behavior**
They aren't.

**Regression**
Bug did not occur in 2.60.1, although we were working from 2.52.1

**Comment by maxnbk** [comment]
Updated the reproduction information with a helpful oneliner to reproduce in bash, provided you have desired package-versions installed and/or tweak the resolve request according to what you have available.

**Comment by nerdvegas** [comment]
Yeah I think this regression is a pure logic fail! I figure we just didn't take into account the case where rez-as-rez-pkg should keep a separate production install of rez visible.
